### PR TITLE
Button component

### PR DESCRIPTION
**Pull request by carlobernardini**
- Button component with tests + story
- Extended variable scheme
- Maps Sass variables to CSS variables

### CODE REVIEW

File: src/components/Button/Button.js [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import React from 'react';
```
Comment (antipin) [inline src/components/Button/Button.js]: Button component should be listed in `src/index.js`

```js
export { default as Dummy } from './components/Dummy';
export { default as Button } from './components/Button';
```

File: stories/Button.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import React from 'react';
+import { storiesOf } from '@storybook/react'; // eslint-disable-line import/no-extraneous-dependencies
+import { boolean, select, text, withKnobs } from '@storybook/addon-knobs'; // eslint-disable-line import/no-extraneous-dependencies
+import Button from '../src/components/Button';
+import { CONTEXTS, SIZES } from '../src/constants';
+
+storiesOf('Button', module)
+    .addDecorator(withKnobs)
+    .add('Button', () => (
+        <Button
```
[the remark this answers is not in the document]
Comment (carlobernardini) [reply stories/Button.js]: Yes, @antipin is looking into a fix

File: src/components/Button/Button.js [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    /** The size of the button */
+    size: PropTypes.oneOf(SIZES),
+    /** whether or not to show block - level button(full width) */
+    isBlock: PropTypes.bool,
+    /** should button be disabled or not */
+    disabled: PropTypes.bool,
+    /** type of the button */
+    type: PropTypes.oneOf(['submit', 'button']),
+    /** optional html attrinutes such as target, href, mouseOver etc. */
+    htmlAttributes: PropTypes.objectOf(
```
Comment (antipin) [inline src/components/Button/Button.js]: I think this one is redundant (also can be applied to `style` and `className`):

We can do something like that:
```js
const Button = props => {
    const { children, disabled, onClick, style, type, block, ...rest } = props;
    return (
        <button
            {...block()}
            type={type}
            onClick={onClick}
            disabled={disabled}
            {...rest}
        >
            {children}
        </button>
    );
};
```

File: src/components/Button/Button.js [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+    type: 'button',
+    htmlAttributes: {},
+    onClick: null,
+    style: null,
+    className: ''
+};
+
+Button.propsToMods = ['context', 'size', 'isBlock'];
+
+export default bem(styles)(Button);
```
[the remark this answers is not in the document]
Comment (carlobernardini) [reply src/components/Button/Button.js]: Good point -- done

File: src/components/Button/Button.js [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+        >
+            {children}
+        </button>
+    );
+};
+
+Button.propTypes = {
+    /** The label of the button */
+    children: PropTypes.oneOfType([PropTypes.string, PropTypes.node]).isRequired,
+    /** The button context (e.g. brand, primary, bad, good etc. - defaults to brand) */
```
[the remark this answers is not in the document]
Comment (carlobernardini) [reply src/components/Button/Button.js]: Yep, fixed

File: src/components/Button/Button.js (hunk 1 above)
Comment (carlobernardini) [reply src/components/Button/Button.js]: Already fixed

File: src/components/Button/Button.js (hunk 3 above)
Comment (carlobernardini) [reply src/components/Button/Button.js]: Good point -- fixed

File: stories/Button.js [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+            context={select('Context', ['link', ...CONTEXTS], CONTEXTS[0])}
+            size={select('Size', SIZES, SIZES[1])}
+            isBlock={boolean('isBlock', false)}
+            type={select('Type', ['submit', 'button'], 'submit')}
+            disabled={boolean('Disabled', false)}
+        >
```
Comment (ribeiroguilherme) [inline stories/Button.js]: Adding `className='my-custom-class'` seems to break the component. Am I doing something wrong here?

**Pull request merged**